test/streams-test.js

```diff
@@ -331,3 +326,71 @@ describe("Streaming sugar", function() {
     });
 });
 
+describe("Encoding using internal modules with surrogates in separate chunks:", function () {
+    // Generate the expected value of surrogates UT
+    function toBufForSurrgates(input) {
+        var str = input.join('')
+        var buf = Buffer.from(str, 'utf8')
+        return buf.toString('hex')
+    }
+
+    it("a single string", checkEncodeStream({
+        encoding: "utf8",
+        input: ["\uD83D\uDE3B"],
+        output: toBufForSurrgates(["\uD83D\uDE3B"])
```

Comment:
you still duplicate the input values here, can we avoid that and use 1 line per test instead of 5?